### PR TITLE
Tweaking the README to sell the `fql` function more

### PR DESCRIPTION
**Pull request by cleve-fauna**
Tweaking the README to sell the `fql` function more.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,6 +118,97 @@ The driver is additionally made available to browsers via CDN:
...
+for example:
+
+```typescript
+let collectionName = "Pets";
+
+// a reusable sub query to determine a collections existence
+let collectionExists = fql`Collection.byName(${collectionName}) != null`;
+
+client.query(fql`
+  // use the sub query we defined elsewhere
```
Comment (cleve-fauna) [inline README.md:137]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -117,6 +118,96 @@ The driver is additionally made available to browsers via CDN:
...
+    Collection.create({ name: ${collectionName} })
+    "Collection exists now!"
+  }
+```
+
+This has several advantages:
+
+- You can use `fql` to build a library of subqueries applicable to your domain - and combinable in whatever way you need
+- injection attacks are not possible if you pass input variables into the interpolated (`${i'm interpoloated}`) parts of the query.
+- the driver speaks "pure FQL X" - you can try out some FQL X on the dashboard's terminal and paste it directly into your app like fql`copied from terminal...` and the query will work as is.
```
Comment (cleve-fauna) [inline README.md:149]:
```suggestion
- the driver speaks "pure FQL X" - you can try out some FQL X on the dashboard's terminal and paste it directly into your app like fql``copied from terminal...`` and the query will work as is.
```

File: README.md [hunk 3]
```diff
@@ -117,6 +118,96 @@ The driver is additionally made available to browsers via CDN:
...
+    Collection.create({ name: ${collectionName} })
+    "Collection exists now!"
+  }
+```
+
+This has several advantages:
+
+- You can use `fql` to build a library of subqueries applicable to your domain - and combinable in whatever way you need
+- injection attacks are not possible if you pass input variables into the interpolated (`${i'm interpoloated}`) parts of the query.
+- the driver speaks "pure FQL X" - you can try out some FQL X on the dashboard's terminal and paste it directly into your app like fql``copied from terminal...`` and the query will work as is.
```
Comment (cleve-fauna) [inline README.md:149]:
```suggestion
- the driver speaks "pure FQL X" - you can try out some FQL X on the dashboard's terminal and paste it directly into your app like fql\`copied from terminal...\` and the query will work as is.
```

**Review by cynicaljoy — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -117,6 +118,96 @@ The driver is additionally made available to browsers via CDN:
...
+    ts,
+    name,
+    email
+  }
+`);
+```
+
+### Advanced Composition example using `fql` function
+
+
```
Comment (ptpaterson) [inline README.md:180]:
```suggestion
### Advanced Composition example using `fql` function

```

Comment (cleve-fauna) [reply README.md:180]: i don't see a difference

Comment (ptpaterson) [reply README.md:180]: I don't think that suggestion renders how I tried to write it...

There's an extra line here, that's all.

Comment (cleve-fauna) [reply README.md:180]: oh one less line

Comment (cleve-fauna) [reply README.md:180]: will get it next time i'm in here. already merged 😮‍💨

Comment (ptpaterson) [reply README.md:180]: Was a tiny nit that doesn't even render :)

**Pull request merged**